src/Infrastructure/Dashboard/TermMeta/TermMetaFieldsValidator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fau\DegreeProgram\Infrastructure\Dashboard\TermMeta;
+
+use WP_Error;
+
+final class TermMetaFieldsValidator
+{
+    public function validate(TermMetaField ...$termMetaFields): ?WP_Error
+    {
+        if (! $this->isTermEditingContext()) {
+            return null;
+        }
+
+        foreach ($termMetaFields as $termMetaField) {
+            $validationPattern = $termMetaField->validationPattern();
+            if (is_null($validationPattern)) {
+                continue;
+            }
+
+            $postedValue = filter_input(
+                INPUT_POST,
+                $termMetaField->key(),
+                FILTER_SANITIZE_SPECIAL_CHARS
+            );
+
+            $sanitizedValue = (string) $termMetaField->sanitize($postedValue);
+
+            // Skip validation if field is empty.
+            if ($sanitizedValue === '') {
+                continue;
+            }
+
+            if (! $validationPattern->matches($sanitizedValue)) {
+                return new WP_Error(
+                    'invalid_term_meta',
+                    sprintf(
+                        // translators: %1$s: field title. %2$s: expected pattern description.
+                        __(
+                            'The value of the field %1$s is invalid. expected value: %2$s',
```

Comment:
Can we wrap the field name in curly quotes (&#8220;,&#8221;)for better readability?
```suggestion
                            'The value of the field &#8220;%1$s&#8221; is invalid. expected value: %2$s',
```